src/http/proxy_wasm/ngx_http_proxy_wasm_dispatch.c

```diff
@@ -606,7 +621,24 @@ ngx_http_proxy_wasm_dispatch_request(ngx_http_proxy_wasm_dispatch_t *call)
 
     b->last = ngx_cpymem(b->last, ngx_http_proxy_wasm_host,
                          sizeof(ngx_http_proxy_wasm_host) - 1);
+
+#if (NGX_HAVE_UNIX_DOMAIN)
+    static ngx_str_t uds = ngx_string("unix:");
+    static ngx_str_t localhost = ngx_string("localhost");
```

Comment:
There is nothing that guarantees `localhost` is a valid hostname, it may not be in `/etc/hosts`. I think we should copy the authority as it was given by the user; the only difference with `unix` socket is that `port` must be set to `0`.